### PR TITLE
Move CodeScene publication to main (CV-005)

### PR DESCRIPTION
**Pull request by leynos**
## What this changes

The merge gate no longer contacts CodeScene. `ci.yml` generates coverage on a
pull request, for its own ratchet check, and carries no CodeScene action, no
`cs-coverage` command and no `CS_ACCESS_TOKEN` anywhere. A new
`coverage-main.yml`, reachable only from a push to `main` or a dispatch,
generates the trunk report, writes the ratchet baseline and uploads it.

That is the estate rule CV-005 (concordat #176, `main-owned-codescene-coverage`).

## Why, and what the trunk runs now

This reverses a decision this repository recorded deliberately, so the reversal
belongs in the open. `coverage-main.yml` was removed once because it duplicated
the Linux lane's executed set on every merge. That objection is answered rather
than accepted: **every** coverage step in `ci.yml` now runs on a pull request
only, so the trunk executes the workspace suite exactly once, here. The merge
gate still runs on a push for its other gates, the publish dry run and its
cache saves; it simply does not test the same commit a second time.

What the earlier reasoning did not price was the cost of putting a third-party
service inside the merge gate. Between 2026-09-16 and 2026-09-18 an unpinned
`cs-coverage` could not parse its own cobertura output. Because the check ran
as a step of the required `build-test` job, every pull request in this
repository was blocked on a failure that had nothing to say about the change
under review and that no pull request could fix. A lane that cannot contact
CodeScene cannot be stopped by CodeScene.

One consequence stated plainly: the trunk no longer executes the Windows lanes'
suite, because this publisher is Linux. Windows is tested on every pull request
against the content that merges, so what is lost is detection of a Windows-only
failure introduced by a semantic merge conflict. The alternative was to leave
the Windows legs testing on a push, which is the duplicate run the rule exists
to remove.

## The two lanes must measure the same thing

Otherwise the baseline the trunk writes is not the baseline a pull request
should be compared against, and a change can fail or pass on the difference
between two measurements rather than on itself.

`tests/workflow_contracts/codescene_coverage_test.py` holds their
`generate-coverage` inputs equal field by field, and holds the list of compared
fields equal to the set both lanes declare, so an input added to both and not
to the list cannot drift unnoticed. It also holds both to one action revision.
`ratchet_publication_test` made that claim about `ci.yml` alone, which was the
whole story while one workflow generated coverage and is not now: the publisher
could be repinned by itself, and the baseline it wrote would then come from a
different implementation than the one measuring against it.

## Pins

Both shared actions move to `a5765019912a8ab6882b12db049c7cde635f3a85`, which
carries the manifest pin that fixed the parse break. The upload passes no
`installer-checksum`: from shared-actions `f68e8e2e` a non-empty value is
rejected outright and `archive-checksum` replaces it, so a repin without the
input change is a red lane rather than a deprecation warning. A contract
refuses the old input anywhere in the repository.

## The rule is proved against documents written for it

Not against `.github/workflows`, which passes a broken scanner as readily as a
working one. The scan clears a workflow by finding nothing, so a marker that
had stopped matching would clear the very thing it exists to catch while every
other marker kept the suite green. Each of the three markers, the action, the
command and the credential, is therefore driven over a document carrying only
its own interaction, and a fourth rule proves the scan can clear as well as
refuse.

## Existing contracts updated, not relaxed

Three encoded the old arrangement:

- `publish_preflight_scope_test` now carries the Linux selector's event clause.
- `runner_placement_test` names both coverage callers, so a third still fails,
  and requires the event clause it previously forbade.

## Mutations

Twenty-eight, each applied alone and undone with the tree verified between, each
detected by a named contract.

| Mutation | Detected by |
| --- | --- |
| the credential returns to the merge gate | `test_no_pull_request_workflow_contacts_codescene` |
| a CodeScene action returns to the merge gate | the same |
| a bare `cs-coverage` command returns | the same |
| the publisher gains a pull-request trigger | that rule and `test_the_publisher_is_not_reachable_from_a_pull_request` |
| the publisher checks instead of uploading | `test_the_publisher_uploads_rather_than_checks` |
| the publisher's push filter widens | `test_the_publisher_is_not_reachable_from_a_pull_request` |
| the deprecated `installer-checksum` returns | `test_no_caller_passes_the_deprecated_installer_checksum` |
| the gate stops declining the artefact | `test_the_pull_request_lane_declines_publication` |
| the Linux lane generates on the trunk too | `test_only_the_publisher_measures_coverage_on_the_trunk` and two placement rules |
| a Windows lane generates on the trunk too | `test_the_merge_gate_runs_no_coverage_on_the_trunk` |
| the trigger reader asks for the string `on` key | `test_the_trigger_reader_survives_the_boolean_on_key` |
| the upload loses its main-ref guard | `test_the_upload_is_restricted_to_the_main_ref` |
| the publisher cancels a running generation | `test_the_publisher_serializes_its_trunk_generations` |
| the group stops reading `github.ref` | the same |
| the two lanes drift on a selection input | `test_both_lanes_measure_the_same_thing` |
| an input is added to both lanes, not to the list | `test_the_selection_list_covers_every_input_both_lanes_declare` |
| the publisher drifts to another revision | `test_both_lanes_call_one_coverage_revision` |
| a pin is loosened to a tag | `test_every_shared_coverage_action_is_sha_pinned` |
| each of the three markers broken in turn | `test_each_marker_finds_its_own_interaction` |
| the scan inverted to report an interaction everywhere | `test_a_workflow_free_of_codescene_is_cleared` |

## Every platform ratchets, and the earlier exception was worse than it looked

An earlier revision of this branch left the two Windows lanes unratcheted and
recorded it as a deliberate platform exception, on the reasoning that comparing
them against a baseline the Linux lane wrote would compare two platforms. That
reasoning was sound about the comparison and wrong about the mechanism.

`generate-coverage` keys its baseline cache by `runner.os`, so a Windows lane
never reads the Linux baseline. It reads a Windows one, and there was none, so
the action's own "Ensure baseline files" step would have written a zero and the
lane would have passed whatever its coverage was. The repository did not have
an exception. It had a ratchet that could not fail, which is worse, because it
looks configured.

Both Windows lanes now ratchet, and `coverage-main.yml` gains a Windows job
that writes the baseline they read. That job uploads nothing: CodeScene reads
the Linux report, so keeping the credential out leaves `coverage-upload` as the
single CodeScene contact. A contract now holds that to one **job** rather than
one workflow, because the upload rules check the job that uploads and neither
would have noticed a second job acquiring the token.

`test_every_ratcheting_platform_has_a_trunk_writer` is the rule this really
needs. It reads each gate lane's platform from its own guard, each publisher
job's from its `runs-on` through a named label table, and holds the two sets
equal in both directions: a lane ratcheting where the trunk never runs passes
against a zero, and a trunk writer nothing reads is a job paying for a baseline
no lane consults.

### What the mechanism cannot give us

The cache key carries `runner.os` and nothing finer, so both Windows lanes read
the one Windows baseline, and the trunk writes it from the default-features
configuration. The strict lane is therefore measured against a figure produced
under a different feature selection. It is the closest the action's keying
allows, and if the strict lane sits below that figure it will say so on the
second trunk run rather than silently. Stated in the developer guide too.

## Two defects the falcon-pachinko round found, present here identically

- **The upload published from any ref.** It was guarded on the credential
  alone. `workflow_dispatch` selects a ref and the push trigger's
  `branches: [main]` says nothing about a dispatch, so a dispatch from a
  feature branch would have published that branch's coverage through the
  main-owned upload. The step now also carries
  `github.ref == 'refs/heads/main'`. The guard sits on the step, not the job,
  so a dispatch still measures the lane end to end.
- **Nothing held the concurrency group.** It was already keyed on the ref with
  `cancel-in-progress: false`, so two pushes to `main` queue rather than race,
  but a later edit could have removed that silently. The action saves a fresh
  baseline cache per successful push and later runs restore the newest match,
  so two overlapping generations would leave the older commit's baseline as the
  one every pull request is measured against.

The publisher-shape rules now live in `codescene_publisher_test.py`.

## The publisher's first run is on main

`workflow_dispatch` only reaches a workflow that exists on the default branch,
so this one cannot be exercised before it merges. Its first run will be the
merge push. Two things reduce the risk, and neither removes it:

- Every step is lifted from `ci.yml`'s Linux path with the matrix references
  replaced by literals, so the environment is the one that lane already passes
  in. The Python half of the coverage run is `testpaths = ["scripts/tests"]`,
  which needs Whitaker for the lint-gate test and none of the Node tooling the
  merge gate installs for Markdown and Mermaid; Whitaker is installed here.
- It reads the same cache keys `ci.yml`'s trunk push writes, so it is warm from
  the first run rather than compiling the workspace cold.

If it does fail, it fails on the trunk. Worth a watch on the merge push rather
than a merge and a walk away.

## Gates

Bare and sequential: `actionlint` on both workflows, `ruff check` and `format`,
`pylint` and the df12 pass, `ambrleaks`, `ty`, `mdtablefix --check`,
`markdownlint-cli2` with 0 errors, the spelling gate, and 218 workflow
contracts, up from 206. No Rust source changed, so the cargo gates were left to
CI.

`actionlint` caught a duplicated `env:` key that `yaml.safe_load` had accepted
silently.

## Summary by Sourcery

Separate pull-request coverage checks from main-owned CodeScene publication and ratchet baseline generation while enforcing consistent, pinned, platform-aware coverage workflows.

New Features:
- Move CodeScene coverage publication and trunk ratchet baseline generation into a dedicated workflow triggered by pushes to main or manual dispatch.
- Add Windows trunk coverage generation so every ratcheting platform has a corresponding baseline writer.

Bug Fixes:
- Prevent CodeScene outages and coverage parsing failures from blocking pull-request merge gates.
- Restrict CodeScene uploads to the main ref and serialize trunk generations to avoid stale baseline publication.
- Replace the deprecated installer checksum input and update shared coverage actions to the corrected pinned revision.

Enhancements:
- Keep pull-request coverage measurement and trunk publication separate while enforcing identical coverage selections across both lanes.
- Strengthen workflow contracts to verify CodeScene isolation, trigger boundaries, action pins, publication behavior, platform coverage, and scanner correctness.
- Enable ratcheting for all pull-request coverage lanes, including Windows, against platform-specific trunk baselines.

CI:
- Limit all ci.yml coverage execution to pull requests and remove CodeScene credentials, commands, and actions from the merge gate.

Documentation:
- Document main-owned CodeScene publication, platform ratchet behavior, baseline limitations, and the resulting workflow responsibilities.

Tests:
- Expand workflow contract coverage with cross-workflow consistency checks, publisher-shape checks, marker fixtures, trigger parsing validation, and mutation-resistant scanner tests.

### CODE REVIEW

File: .github/workflows/coverage-main.yml [hunk 1]
```diff
@@ -0,0 +1,687 @@
...
+          {
+            printf '### Resources\n\n'
+            printf -- '- peak used memory: %s MiB\n' "$peak_memory"
+            printf -- '- peak used disk: %s MiB\n' "$peak_disk"
+            printf -- '- least free disk: %s MiB\n' "$least_free"
+            printf -- '- samples: %s at 15 second intervals\n' "$count"
+            printf -- '- shape: %s vCPU\n' "${RUNNER_VCPUS:-unknown}"
+          } >> "$GITHUB_STEP_SUMMARY"
+
+  coverage-baseline-windows:
```
[the remark this answers is not in the document]
Comment (leynos) [reply .github/workflows/coverage-main.yml:460]: Answered in two parts, one argued and one fixed.

The length itself: the 400-line rule in `AGENTS.md` binds code modules and is enforced by the second Pylint pass's `max-module-lines` over the Python sources. Workflows have never been held to it here; `ci.yml` is 888 lines on `main`. Splitting `coverage-main.yml` into two workflow files would not shorten the Linux job, which is itself over 400 lines, and a composite action for the shared setup would hide those steps from the contracts in `tests/workflow_contracts`, which read each workflow's own steps.

The drift risk is real, and it is now asserted rather than trusted. `tests/workflow_contracts/coverage_publisher_setup_test.py`:

- holds each publisher job's `env` to the merge gate's job `env`, key for key and value for value (only `RUST_TOOLCHAIN` and `CS_ACCESS_TOKEN` may be extra);
- requires every publisher job to repeat the named setup steps, so deleting one fails rather than leaving one fewer comparison;
- holds every repeated step equal to its `ci.yml` counterpart after resolving `${{ matrix.* }}` against the leg the job mirrors and `${{ env.* }}` against each job's own `env`, ignoring only names, guards and comment lines inside `run:` bodies.

Proved by mutation: moving the Windows publisher's `SCCACHE_VERSION` to `0.11.0` fails the pin rule, and changing its `SCCACHE_CACHE_KEY` schema to `v3` fails the step rule.

File: docs/developers-guide.md [hunk 2]
```diff
@@ -146,20 +147,80 @@ already enables it and a separate Linux lane executed nothing new. The Windows
...
+service inside the merge gate. Between 2026-09-16 and 2026-09-18 an unpinned
+`cs-coverage` could not parse its own cobertura output, and because the check
+ran as a step of the required `build-test` job, every pull request in this
+repository was blocked on a failure that had nothing to say about the change
+under review, and that no pull request could fix. A lane that cannot contact
+CodeScene cannot be stopped by CodeScene.
+
+One consequence to state plainly: the trunk no longer executes the Windows
+lanes' suite at all, because this publisher is Linux. Windows is tested on
+every pull request, against the same content that merges, so what is lost is
```
[the remark this answers is not in the document]
Comment (leynos) [reply docs/developers-guide.md]: Fixed. The stale sentence is gone from `docs/developers-guide.md`, "CodeScene publication belongs to main". The guide now says the trunk runs each platform's suite once, in `coverage-main.yml`: the Linux suite in `coverage-upload` and the Windows default-features suite in `coverage-baseline-windows`. What the trunk no longer runs is the strict-validation Windows lane, which is pull-request only, so the loss is narrower than the old sentence claimed: a failure specific to that feature selection on Windows, introduced by a semantic merge conflict.

File: .github/workflows/coverage-main.yml [hunk 3]
```diff
@@ -0,0 +1,687 @@
...
+      # different Cargo generations.
+      RUST_TOOLCHAIN: stable-x86_64-pc-windows-msvc
+    steps:
+      # GitHub-hosted Windows images ship Git, Git Bash, and Chocolatey but
+      # not GNU Make, which `make publish-check` needs on every lane.
+      - name: Install GNU Make
+        shell: pwsh
+        run: |
+          choco install make --yes --no-progress
+          Get-Command make | Out-Null
```
[the remark this answers is not in the document]
Comment (leynos) [reply .github/workflows/coverage-main.yml]: @coderabbitai Fixed in `.github/workflows/coverage-main.yml`, `coverage-baseline-windows`:

- The duplicated "Stated once here," line is gone; the `RUST_TOOLCHAIN` comment reads once.
- The GNU Make comment no longer claims a `make publish-check` this job never runs. It says the image lacks Make and that the measured suite can invoke `make lint-whitaker` through the `trybuild_macros` Whitaker test when `cargo dylint` and `whitaker` are available, so installing it keeps the measured environment the pull-request lane's.
- The same stale `make publish-check` claim was also on this job's checkout step. That comment now says the coverage action reads the history.
- Two stacked comment blocks on `Restore compiler cache` were merged into one, which also says the step's archive is now written by this job.

File: .github/workflows/coverage-main.yml [hunk 4]
```diff
@@ -0,0 +1,687 @@
...
+      - name: Restore compiler cache
+        id: sccache
+        # Windows has no Actions-cache sccache backend, because the shared
+        # Rust setup is called with `use-sccache: false`, so it always uses
+        # the local directory this step restores.
+        uses: actions/cache/restore@55cc8345863c7cc4c66a329aec7e433d2d1c52a9  # v6.1.0
+        with:
+          path: ${{ github.workspace }}/.sccache
+          key: ${{ env.SCCACHE_CACHE_KEY }}
+          restore-keys: ${{ env.SCCACHE_CACHE_RESTORE_PREFIX }}
```
[the remark this answers is not in the document]
Comment (leynos) [reply .github/workflows/coverage-main.yml:629]: @coderabbitai Implemented as proposed, with one extension.

- `ci.yml` no longer writes the compiler cache. The step is replaced by a comment saying why: that lane's trunk run no longer builds the instrumented workspace, so an archive written there would hold only the publish dry run's objects.
- `coverage-main.yml:coverage-baseline-windows` saves `.sccache` after its coverage build, guarded `github.event_name == 'push' && github.ref == 'refs/heads/main' && steps.sccache.outputs.cache-hit != 'true'`.
- The extension: `coverage-upload` gets the same writer for the Linux lanes' `RSTEST_BDD_SCCACHE_LOCAL` fallback, guarded on that variable as well, because the removed `ci.yml` step was that mode's writer too.
- `SCCACHE_CACHE_KEY` and its restore prefix move from `sccache-v1-` to `sccache-v2-` in both workflows, for the reason you gave: the `v1` archive for the current lockfile is immutable.

Held by `tests/workflow_contracts/coverage_publisher_setup_test.py`: the save steps keyed on `${{ env.SCCACHE_CACHE_KEY }}` across the whole repository are exactly the two publisher jobs, and each writer's guard is evaluated, not searched, for a trunk push that missed (saves), a trunk push that hit, a dispatch from a branch and a dispatch on `main` (none save). Mutations: restoring the `ci.yml` writer fails the one-writer rule; dropping the event and ref clauses from the Windows writer fails both dispatch cases.

The cost is stated in `docs/developers-guide.md`, "Compiler cache": the publish dry run's package-verification objects are no longer persisted for the pull-request Windows lanes. The instrumented all-targets build is the larger of the two workloads, and it is what both Windows lanes and the trunk's own Windows job now restore.

File: docs/developers-guide.md [hunk 5]
```diff
@@ -146,20 +147,80 @@ already enables it and a separate Linux lane executed nothing new. The Windows
...
+a ratchet that could not fail, which is worse, because it looks configured.
+
+`coverage-main.yml` therefore runs a Windows job as well as the Linux one. It
+writes the Windows baseline and uploads nothing: CodeScene reads the Linux
+report, so keeping the credential out of that job leaves `coverage-upload` as
+the single CodeScene contact in the repository. A contract holds the two sets
+equal, so a lane that ratchets on a platform the trunk does not run fails here
+rather than passing against a zero.
+
+One thing the mechanism cannot give us. The cache key carries `runner.os` and
```
[the remark this answers is not in the document]
Comment (leynos) [reply docs/developers-guide.md]: @coderabbitai Fixed: the sentence in `docs/developers-guide.md` now reads "One thing the mechanism cannot give." with the rest of the paragraph unchanged.

File: tests/workflow_contracts/codescene_coverage_support.py [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+    MissingStepInputsError
+        If the step declares no inputs.
+    """
+    for reference in iter_steps(workflow_name):
+        if reference.name != step_name:
+            continue
+        inputs = reference.step.get("with")
+        if not isinstance(inputs, dict):
+            raise MissingStepInputsError(str(reference))
+        return inputs
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/workflow_contracts/codescene_coverage_support.py:304]: @coderabbitai Fixed, and made stricter than the suggestion. `coverage_step` in `tests/workflow_contracts/codescene_coverage_support.py` now takes `job_name` as a required argument rather than an optional one, so no caller can fall back to the first match by omission, and a missing step's error names the job and the steps that job declares. Every call site passes its job: the publisher contracts pass `PUBLISHER_JOB` or the pair's publisher job, and the gate contracts pass `build-test`.

Proved by mutation: making the lookup ignore the job again fails `test_each_lane_measures_what_its_baseline_writer_measures` for the Windows pair, because the Windows lookup then returns the Linux job's step.

File: tests/workflow_contracts/codescene_coverage_test.py [hunk 7]
```diff
@@ -1,225 +1,322 @@
...
+    difference rather than on its own change.
+    """
+    gate = coverage_step(PR_WORKFLOW, PR_COVERAGE_STEP)
+    publisher = coverage_step(PUBLISHER, PUBLISHER_COVERAGE_STEP)
+
+    assert gate.get(field) == publisher.get(field), (
+        f"{PR_COVERAGE_STEP} selects {field}={gate.get(field)!r} and "
+        f"{PUBLISHER_COVERAGE_STEP} selects {publisher.get(field)!r}; the "
+        "baseline would not be comparable with what the ratchet checks"
+    )
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/workflow_contracts/codescene_coverage_test.py]: @coderabbitai Implemented. `test_each_lane_measures_what_its_baseline_writer_measures` in `tests/workflow_contracts/codescene_coverage_test.py` is parametrized over `PAIRS` in the new `tests/workflow_contracts/coverage_lane_pairs.py`: the Linux lane with `coverage-upload`, and `Test and Measure Coverage (Windows, default features)` with `coverage-baseline-windows`.

The gate lane's `${{ matrix.* }}` references are resolved against the matrix row its own `if:` admits on a pull request, evaluated rather than typed beside it, and the rule fails unless exactly one row is admitted. Every input either side declares is then compared, not a named list, so `features`, `with-default-features` and `use-cargo-nextest` are covered along with anything added later to one side only. Only `publish-artefact` is excluded, because every pull-request lane now declines the artefact and the publisher keeps the default.

Proved by mutation: switching the Windows baseline writer to `use-cargo-nextest: true` fails the Windows pair.

**Comment by leynos** [comment]
@coderabbitai Have the following failed checks now been resolved at the current head, 429f79f00eebe47d035c61a2b6a374d08428016e? If further work is required, please provide an AI agent prompt for the remaining work.

**Testing (Overall).** Each gap named in the resolution now has a contract:

- The publisher's triggers must be exactly `push` and `workflow_dispatch`, with `push.branches == ['main']`. Triggers are read in scalar, sequence and mapping form under either key (`codescene_upload_test.py`, `pull_request_reach.py`).
- The upload step's inputs are asserted whole: `path: coverage.xml`, `format: cobertura`, `mode: upload`, `access-token: ${{ env.CS_ACCESS_TOKEN }}`.
- The credential's exact locations in the publisher are asserted as a set: the upload job's `env`, the upload step's guard, and its `access-token` input. Moving it to workflow `env` or onto the coverage step fails. Across everything a pull request can run, the scan covers job, step, input, `run` and `secrets:` scope, including `secrets: inherit`, over the closure of same-repository reusable-workflow calls.
- The Windows default-features gate lane is compared with `coverage-baseline-windows` on every input either side declares, after resolving matrix references against the row the lane's own guard admits. Ratchet and toolchain equality were already held.
- Every publisher job's `env` pins, setup steps and repeated step bodies are held equal to `ci.yml:build-test`'s (`coverage_publisher_setup_test.py`). The compiler-cache writer is held to one per platform.

**Testing (Unit And Behavioural).** This repository has no workflow runner. GitHub evaluates `if:` expressions, resolves `secrets`, and serves the Actions cache and the Ubicloud proxy, and none of those can be executed locally with fidelity: `act` cannot host the Windows job at all. The behavioural question the row asks is answered instead by evaluating each guard as GitHub would (`guard_conditions.py`): conjunctions of equality comparisons, case-insensitive, refusing `||`, grouping and negation, and refusing any context value the caller did not state.

- `codescene_upload_test.py` evaluates the upload guard for a push to `main`, a dispatch on `main`, a dispatch from a branch, a dispatch from a tag, and a missing token.
- `test_the_merge_gate_runs_no_coverage_on_the_trunk` evaluates every gate coverage step for every matrix leg on a push to `main`.
- `test_the_compiler_cache_is_written_only_from_a_trunk_push` evaluates each cache writer for a trunk miss, a trunk hit, and dispatches.
- The ratchet baseline is saved only under the shared action's `publish-baseline: auto`, and a contract refuses any publisher job widening it.

The error paths the row names now have direct cases: a list-form trigger block refused where a mapping is needed, an absent coverage step naming its job, and a step with no inputs (`codescene_scanner_property_test.py`). That module also adds the requested Hypothesis properties for `references_in`. The first live proof of the publisher itself is its first trunk run after merge, which will be read and reported.

**Comment by leynos** [comment]
@coderabbitai Your previous reply confirmed the contract gaps resolved at 429f79f00eebe47d035c61a2b6a374d08428016e, pending the three `build-test` jobs. Have the failed checks now been resolved at the current head, e9643c0612a3dfdb78d96e6148c9855cac986ecb? If further work is required, please provide an AI agent prompt for the remaining work.

All three required `build-test` jobs have passed at this head. Three commits have landed since your reply:

- **858c2df3** drives `pull_request_workflows` over a call chain. The closure was proved on its own, but the query the boundary rule calls ran only over this repository's files, which contain no local call, so a fallback to the trigger list survived every case.
- **b832f21c** holds every `upload-codescene-coverage` call to a revision verified to carry the CLI manifest, and refuses a `get-codescene-sha.yml` refresher even when it names no variable.
- **e9643c06** follows GitHub's documented `$/` self-repository spelling and refuses `$/...@ref`. The guard reader now refuses a stray `)`. The publisher's concurrency wording now says exactly what the group guarantees: no overlap and no cancelled running generation. A newer push replaces an older pending run.

Each new clause fails under its own mutation.
